File: app/src/main/java/lsl/myapplication/MainActivity.java
package lsl.myapplication;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;

import lsl.myapplication.loading.CustomLoading;

public class MainActivity extends AppCompatActivity {

    CustomLoading mCustomLoading;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
//        CustomView customView= (CustomView) findViewById(R.id.test);
//        PieData pieData1=new PieData();
//        pieData1.setPercentage(0.2f);
//
//        PieData pieData2=new PieData();
//        pieData2.setPercentage(0.4f);
//
//
//        PieData pieData3=new PieData();
//        pieData3.setPercentage(0.1f);
//
//        PieData pieData4=new PieData();
//        pieData4.setPercentage(0.3f);
//
//
//        List<PieData> pieDatas=new ArrayList<>();
//        pieDatas.add(pieData1);
//        pieDatas.add(pieData2);
//        pieDatas.add(pieData3);
//        pieDatas.add(pieData4);
//
//        customView.setDatas(pieDatas);


        mCustomLoading = (CustomLoading) findViewById(R.id.loading);

    }


    public void doClick(View v) {
        mCustomLoading.setProcess();
    }


}

File: recycleldviewdemo/src/main/java/com/lsl/recycleldview/MainActivity.kt
package com.lsl.recycleldview

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import kotlinx.android.synthetic.main.activity_main.*


class MainActivity : AppCompatActivity() {
    lateinit var leftAdapter:LeftAdapter
    lateinit var rigthAdapter:RigthAdapter
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        recycle_left.layoutManager=LinearLayoutManager(this)
        leftAdapter= LeftAdapter()
        recycle_left.adapter=leftAdapter


        recycle_rigth.layoutManager=LinearLayoutManager(this)
    }






    class LeftAdapter :RecyclerView.Adapter<LeftHolder>(){
        override fun onBindViewHolder(holder: LeftHolder?, position: Int) {
        }

        override fun onCreateViewHolder(parent: ViewGroup?, viewType: Int): LeftHolder {
           return LeftHolder(itemView =LayoutInflater.from(parent?.context).inflate(R.layout.item_left,parent,false) )
        }

        override fun getItemCount()=20

    }



    class RigthAdapter : RecyclerView.Adapter<RigthHolder>() {
        override fun onCreateViewHolder(parent: ViewGroup?, viewType: Int): RigthHolder {
            return RigthHolder(itemView =LayoutInflater.from(parent?.context).inflate(R.layout.item_rigth,parent,false) )
        }

        override fun onBindViewHolder(holder: RigthHolder?, position: Int) {
        }

        override fun getItemCount()=20

    }












    class LeftHolder(itemView: View) :RecyclerView.ViewHolder(itemView)
    class RigthHolder(itemView: View) :RecyclerView.ViewHolder(itemView)

}